src/otx/core/model/entity/visual_prompting.py

```diff
@@ -838,11 +953,19 @@ def _calculate_mask_iou(mask1: np.ndarray, mask2: np.ndarray) -> np.ndarray:
             overlapped_label = []
             overlapped_other_label = []
             for (im, mask), (jm, other_mask) in product(enumerate(masks), enumerate(other_masks)):
-                if _calculate_mask_iou(mask, other_mask) > threshold_iou:
+                _mask_iou = _calculate_mask_iou(mask, other_mask)
+                if _mask_iou > threshold_iou:
                     if used_points[label][im][2] > used_points[other_label][jm][2]:
                         overlapped_other_label.append(jm)
                     else:
                         overlapped_label.append(im)
+                elif _mask_iou > 0:
+                    # refine the slightly overlapping region
+                    overlapped_coords = np.where(np.logical_and(mask, other_mask))
```

Comment:
Why does here `_inspect_overlapping_areas()` looks same as the one in src/otx/algo/visual_prompting/zero_shot_segment_anything.py?

Reply:
torch version and numpy version. It is needed to merge two implementations as you said, but it is out of scope of this PR. I'll touch it next time.